mitodl/open-discussions
# Handle unicode chars when truncating file content

#### Pre-Flight checklist



#### What are the relevant tickets?
Closes #3975 

#### What's this PR do?
Avoids truncating unicode characters halfway through

#### How should this be manually tested?
Set `ELASTICSEARCH_MAX_REQUEST_SIZE=500000` and also copy values for `MITX_ONLINE_*` from production to your .env file.

Run the following on the master branch and this PR branch.  It should raise a JSONDecodeError on master branch but not this one.  You may need to run `manage.py backpopulate_mitxonline_data` and `manage.py backpopulate_mitxonline_files` first if you haven't already done so.

```
from course_catalog.models import LearningResourceRun
from search.indexing_api import index_run_content_files

run = LearningResourceRun.objects.get(run_id="course-v1:MITxT+14.73x+3T2023")
index_run_content_files(run.id)
```
